src/boot.c

```diff
@@ -687,8 +690,24 @@ void write_volume_label(DOS_FS * fs, char *label)
     if (de.name[0] == 0xe5)
 	de.name[0] = 0x05;
 
-    now = time(NULL);
-    mtime = (now != (time_t)-1) ? localtime(&now) : NULL;
+    source_date_epoch = getenv("SOURCE_DATE_EPOCH");
+    if (source_date_epoch) {
+        char *tmp = NULL;
+        long long conversion = 0;
+        errno = 0;
+        conversion = strtoll(source_date_epoch, &tmp, 10);
+        now = conversion;
+        if (!isdigit(*source_date_epoch) || *tmp != '\0' || errno != 0
```

Comment:
And now I see there another issue. `*source_date_epoch` is of type `char`, which means `signed char` on x86, so contains values from range `[-128, 127]` (values `[128, 255]` are encoded as negative). But function `isdigit()` takes type `int` and expects characters to be in range `[0, 255]`, so this function is incompatible with `signed char` type.

So explicit cast to `unsigned char` is required prior passing value into `isdigit()`. This trap is documented also in manpage, in NOTES section: https://man7.org/linux/man-pages/man3/isdigit.3.html

```suggestion
        if (!isdigit((unsigned char)*source_date_epoch) || *tmp != '\0' || errno != 0
```

Reply:
@pali: Thanks! I updated the 3 "isdigit()" calls with that cast.

Reply:
I fixed this trap also for other is* functions.